### DISCUSSION TITLE
Add a feature to show turns ahead

### DISCUSSION THREAD

**Discussion by TheAdventurer64** [organicmaps/organicmaps#1320] [Ideas]
Having a feature to show turns ahead would be great, as it would help give drivers a heads up on the next few turns, and would help Organic Maps be consistent with other big navigation apps, such as Google Maps, Waze, and Apple Maps. Here's an example of what it could look like:![image](https://user-images.githubusercontent.com/50059322/133914315-2c8e2689-a3b7-4720-ad31-673e1cc81161.png)

**Comment by bahbarnett** [comment]
This seems immensely useful.

On long trips, I look at a route, and often see the next change is 300km away.  So I memorize the change, and save battery.

But even just showing the next turn is so helpful.  On a highway, signs might show "Blah blah street, turn off in  7km". 

When a road is 8 lanes wide per side, one often needs to start moving over well before the map program prompts.  If I know the next turn ahead, I can keep an eye out, and move over in preparation.

**Comment by TravisBrace** [comment]
Its so scary going on a long trip and not knowing which way I'm going or what to look out for. The directions don't show until about half a mile away. Driving through NYC or something is so anxiety inducing. 

Would love to help with this

**Reply by markcellus** [reply]
Yeah I had to switch to Here WeGo for city driving unfortunately. Maybe that can help you in the meantime as well.

**Comment by bahbarnett** [comment]
It feels very strange to me that there's zero interest in this.  It is a major show stopper, making Organic Maps utterly useless for driving in many circumstances.  It seems to me that none of the core team drives all that much, which is fair, but unhelpful in this case.  No dogfooding.

I literally cannot ever use OrganicMaps as a result of this.  Hints to turn are never provided soon enough, sometimes even for rural roads.  Sad.

**Reply by biodranik** [reply]
Your assumptions are wrong. We drive. And seeing directions would be better than not seeing them. But there are [almost 1900 other issues](https://github.com/organicmaps/organicmaps/issues/) that are also important and require our attention and time. Which is extremely limited for a free-time, donations-based project. Plus maintaining GitHub, providing support, testing, and publishing app releases and news.

Don't blame. Don't complain. It's more efficient [to help and support](https://organicmaps.app/support-us/) the project. Any help is appreciated!

**Comment by bahbarnett** [comment]
I find it hard to believe the core dev team drives much, and dogfoods, when OrgranicMaps cannot be used in that way reliably.  Certainly you cannot drive on any freeway, and even driving on city streets is essentially unusable.  It's a core feature that people I've shown it to, also feel is a deal breaker.

Knowing what is a priority, or knowing when something makes a project completely unusual for a large swath of the people who might use that project, is valuable feedback.  I'd say "being able to even hope to use OrganicMaps at all", is probably higher value than almost all of those 1900 bugs, many of which are "can you tweak this little one thing".

And on the linked url, I see this:

"Our small team is very grateful for your feedback and support. Organic Maps would not be possible without our users."

You shouldn't take my response as "complaining", it's not meant that way.  So if you have, then please don't.  It's feedback, which this project is supposedly very grateful for.  And it's aimed at a core-thing missing, which makes Organic Maps unusable for many people, which sort of ties into the "users" mentioned on the second sentence.

And yes, I don't expect anyone to drop all 1900 bugs, or prioritize what I'm saying here above those bugs just because I say so.  Instead, you should consider this feedback from a user that wants this project to succeed, despises Google, and Google Maps and its tracking, but sees it blocked by one small thing.

And showed up after commenting on this bug two years ago, and was just sad that the tyranny of Google Maps still reigns all mighty in my life.

**Reply by Misalf-git** [reply]
Expressing frustation about a missing feature is certainly all right, as it is feedback and may influence its prioritisation.
Beyond that, to complain about priorities or demand considerations is probably rather not going to prompt swift willingness to serve satisfaction.
I don't believe there would be any aversions to this feature, so while the app is utterly unusable currently, some time in the future you might come back and "_now_ they have it".
